libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_Sockets.c

```diff
@@ -1439,11 +1442,10 @@ FreeRTOS_Socket_t *pxSocket;
 		#endif /* ipconfigUDP_MAX_RX_PACKETS */
 
 		case FREERTOS_SO_UDPCKSUM_OUT :
-			/* Turn calculating of the UDP checksum on/off for this socket. */
-			/* The expression "pvOptionValue" of type "void const *" is cast to type "BaseType_t". */
-			lOptionValue = ipNUMERIC_CAST( BaseType_t, pvOptionValue );
+			/* Turn calculating of the UDP checksum on/off for this socket. If pvOptionValue
+			 * is anything else than NULL, the checksum generation will be turned on. */
 
-			if( lOptionValue == 0 )
+			if( pvOptionValue == NULL )
```

Comment:
Genius!
Here a pointer is treated as a number. There is no need to cast from pointer to integer, a NULL test is enough.